csharp/ql/test/resources/stubs/System.Web.cs

```diff
@@ -178,6 +178,37 @@ public class HttpRequest
         public string RawUrl { get; set; }
         public HttpCookieCollection Cookies => null;
         public bool IsAuthenticated { get; set; }
+        public NameValueCollection Form => null;
+        public NameValueCollection Headers => null;
+        public NameValueCollection Params => null;
+        public string UserAgent(string s) => null;
+        public string UrlReferrer(string s) => null;
+        public NameValueCollection ServerVariables => null;
+        // Default property that goes through the collections
+        //      QueryString, Form, Cookies, ClientCertificate and ServerVariables
+        public String this[String key]
+        {
+            get
+            {
+                String s;
+
+                s = QueryString[key];
+                if (s != null)
+                    return s;
+
+                s = Form[key];
+                if (s != null)
+                    return s;
+
+                HttpCookie c = Cookies[key];
+                if (c != null)
+                    return c.Value;
+
+                s = ServerVariables[key];
+                if (s != null)
+                    return s;
+            }
+        }
```

Comment:
The logic itself shouldn't be included in the stub implementation. If you have any logic relying on the flow from other properties, this should be captured by models as data models instead.